x/auth/ante/basic.go

```diff
@@ -200,3 +200,32 @@ func (txh TxTimeoutHeightDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simul
 
 	return next(ctx, tx, simulate)
 }
+
+type TxSigBlockHeightDecorator struct {
+	ak AccountKeeper
+}
+
+func NewTxSigBlockHeightDecorator(ak AccountKeeper) TxSigBlockHeightDecorator {
+	return TxSigBlockHeightDecorator{
+		ak: ak,
+	}
+}
+
+func (txs TxSigBlockHeightDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
+	if !simulate {
+		params := txs.ak.GetParams(ctx)
+		sbh := tx.GetSigBlockHeight()
+		current := uint64(ctx.BlockHeight())
+		validMin := uint64(0)
```

Comment:
Is it ok even if block start height is not 1 by specifying `initalHeight`?

Reply:
For example, if start height is 50 and `ValidSigBlockPeriod` is 100 and current height is 80, then `sig block height` 40 is valid according to the code above. I think the `sig block height` lower than start height has no problem.